module/module.go

```diff
@@ -464,16 +641,19 @@ type Instance struct {
 
 // Dependency defines a module instance required by another module instance
 type Dependency struct {
-	ClassName    string
+	// ClassName is the name of the class as defined in the module system
+	ClassName string
+	// InstanceName is the name of the dependency instance as configu
 	InstanceName string
 }
 
 // JSONClassConfig maps onto a json configuration for a class type
 type JSONClassConfig struct {
-	Name         string              `json:"name"`
-	Config       interface{}         `json:"config"`
-	Dependencies map[string][]string `json:"dependencies"`
-	Type         string              `json:"type"`
+	Name              string              `json:"name"`
+	Config            interface{}         `json:"config"`
+	Dependencies      map[string][]string `json:"dependencies"`
+	Type              string              `json:"type"`
+	IsExportGenerated *bool               `json:"IsExportGenerated"`
```

Comment:
Because defaults are not the same, I couldn't find a way to configure the default for missing boolean values, and I think this is better than IsExportCustom (just because generated is clearer in what it means).